docs/src/main/asciidoc/redis.adoc

```diff
@@ -0,0 +1,530 @@
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `redis-quickstart` {quickstarts-tree-url}/redis-quickstart[directory].
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=redis-quickstart \
+    -Dextensions="vertx-redis, resteasy-jsonb, resteasy-mutiny"
+cd redis-quickstart
+----
+
+This command generates a Maven project, importing the Redis extension.
+
+== Starting the Redis server
+
+Then, we need to start a Redis instance (if you do not have one already) using the following command:
+
+[source, bash]
+----
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name redis_quarkus_test -p 6379:6379 redis:5.0.6
+----
+
+== Configuring Redis properties
+
+Once we have the Redis server running, we need to configure the Redis connection properties.
+This is done in the `application.properties` configuration file. Edit it to the following content:
+
+[source]
+----
+quarkus.redis.hosts=localhost:6379 <1>
+----
+
+1. Configure Redis hosts to connect to. Here we connect to the Redis server we started in the previous section
+
+== Creating the Increment POJO
+
+We are going to model our increments using the `Increment` POJO.
+Create the `src/main/java/org/acme/redis/Increment.java` file, with the following content:
+
+[source, java]
+----
+package org.acme.redis;
+
+public class Increment {
+    public String key; <1>
+    public int value; <2>
+
+    public Increment(String key, int value) {
+        this.key = key;
+        this.value = value;
+    }
+
+    public Increment() {
+    }
+}
+----
+
+1. The key that will be used as the Redis key
+2. The value held by the Redis key
+
+
+== Creating the Increment Service
+
+We are going to create an `IncrementService` class which will play the role of a Redis client.
+With this class, we'll be able to do the perform the `SET`, `GET` , `DELET`, `KEYS` and `INCRBY` Redis commands.
+
+Create the `src/main/java/org/acme/redis/IncrementService.java` file, with the following content:
+
+[source, java]
+----
+package org.acme.redis;
+
+import io.quarkus.vertx.redis.SyncRedisAPI;
+import io.smallrye.mutiny.Uni;
+
+import io.vertx.mutiny.redis.client.Response;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+class IncrementService {
+
+    @Inject
+    SyncRedisAPI syncRedisAPI; <1>
+
+    @Inject
+    io.vertx.mutiny.redis.client.RedisAPI mutinyRedisAPI; <2>
```

Comment:
Do we really want the FQCN here?

Reply:
Surely not. I’ll update this part along with the guide.